routes/problems_id.js

```diff
@@ -61,7 +61,7 @@ module.exports = function(app) {
   });
 
   app.post('/problems/:id', async function(request, response) {
-    const user  = await auth.getUser(request.cookies.token);
+    const user  = (await auth.getUser(request.cookies.token)).user;
```

Comment:
this seems like it might trigger a null dereference since the promise can resolve to null.

Reply:
thank you, fixed